### PR TITLE
small refactor of quantum_initializer

### PR DESCRIPTION
**Pull request by 1ucian0**
Some of the files in `quantum_initializer` create functions around instead of static methods. This PR moves those functions to methods.

### CODE REVIEW

File: qiskit/extensions/quantum_initializer/ucg.py [hunk 1]
```diff
@@ -244,20 +245,20 @@ def _demultiplex_single_uc(self, a, b):
...
 
+    @staticmethod
+    def _ct(m):
+        return np.transpose(np.conjugate(m))
 
-def _ct(m):
-    return np.transpose(np.conjugate(m))
+    @staticmethod
+    def _h():
+        return 1 / np.sqrt(2) * np.array([[1, 1], [1, -1]])
```
Comment (ajavadia) [inline qiskit/extensions/quantum_initializer/ucg.py]: this method should just be deleted. all uses can be replaced by `HGate().to_matrix()`
where HGate is in the standard library of gates.

File: qiskit/extensions/quantum_initializer/ucg.py [hunk 2]
```diff
@@ -244,20 +245,20 @@ def _demultiplex_single_uc(self, a, b):
...
-
-def _h():
-    return 1 / np.sqrt(2) * np.array([[1, 1], [1, -1]])
-
-
-def _rz(alpha):
-    return np.array([[np.exp(1j * alpha / 2), 0], [0, np.exp(-1j * alpha / 2)]])
+    @staticmethod
+    def _rz(alpha):
+        return np.array([[np.exp(1j * alpha / 2), 0], [0, np.exp(-1j * alpha / 2)]])
```
Comment (ajavadia) [inline qiskit/extensions/quantum_initializer/ucg.py]: similar for this one, delete the static method and replace uses by `RzGate(alpha).to_matrix()`

Comment (1ucian0) [reply qiskit/extensions/quantum_initializer/ucg.py]: `qiskit.extensions.standard.rz.RZGate` does not have a `to_matrix` yet. I can create an issue to track that so we can move forward with this PR independently.

File: qiskit/extensions/quantum_initializer/ucg.py (hunk 1 above)
Comment (1ucian0) [reply qiskit/extensions/quantum_initializer/ucg.py]: done

File: qiskit/extensions/quantum_initializer/ucg.py (hunk 2 above)
Comment (1ucian0) [reply qiskit/extensions/quantum_initializer/ucg.py]: Submitted in #3304

Comment (kdk) [reply qiskit/extensions/quantum_initializer/ucg.py]: I thought there was a motivation in leaving out some of the `to_matrix` methods in that their textbook definitions would differ from what was returned from the operator by a global phase.

e.g.
```
>>> Operator(RZGate(1.5)).data
array([[1.       +0.j        , 0.       +0.j        ],
       [0.       +0.j        , 0.0707372+0.99749499j]])
>>> _rz(1.5)
array([[0.73168887+0.68163876j, 0.        +0.j        ],
       [0.        +0.j        , 0.73168887-0.68163876j]])
```

Comment (ajavadia) [reply qiskit/extensions/quantum_initializer/ucg.py]: yeah this is a bigger discussion about global phases and chris has is preparing an RFC for what qiskit convention should look like.. so won't block this PR.

Comment (1ucian0) [reply qiskit/extensions/quantum_initializer/ucg.py]: Maybe a discussion to have in #3304  @kdk  ?

**Pull request merged**